src/aframe_components/custom-components.ts

```diff
@@ -0,0 +1,87 @@
+import { Vector3 } from 'three';
+import { Component } from 'aframe';
+
+const HAPTIC_RANGE = 0.3;
+const COLLISION_DISTANCE = 0.029;
+const HAPTIC_EXPONENT = 2;
+const HAPTIC_GROWTH_MODIFIER = 4;
+const HAPTIC_DURATION = 5000;
+const INTERVAL_DURATION = 100;
+
+interface State {
+  currInterval: number;
+  currPosition: Vector3;
+  distances: Vector3[];
+  currDistance: number;
+  currDistanceScaled: number;
+  intensity: number;
+}
+
+export interface Guidance extends Component {
+  state: State;
+  pointPositions: Vector3[];
+}
+
+export const guideComponent = {
```

Comment:
`export const guideComponent: Guidance = {`
I think this will mean less casting when you access `state` or `pointPositions`

Reply:
I changed to exporting with the Guidance type, which required adding the properties of the component type (which my guidance interface extends) to the guidance component definition.

I did this by including the properties of the component as follows:
el: this.el,
data: this.data

In order to maintain the components default values while including an explicit definition of them in the guidance component. Is this a good way to handle this?

Reply:
I tried this, but I think guideComponent isn't of type Guidance. It is used as the `description` property when I call AFRAME.registerComponent in maine.ts and test.ts. registerComponent return the component, which is of type guidance. But, when I'm accessing the component using entity.components.guide in my testing, it is still being returned as a generic component, which I then cast as type Guidance in order to access its properties. I'm not sure how to get around this.